src/dos/dos_files.cpp

```diff
@@ -603,6 +603,135 @@ bool DOS_CreateFile(const char* name, FatAttributeFlags attributes,
 	}
 }
 
+enum class FileSharingMode
+{
+	Compatibility,
+	DenyReadWrite,
+	DenyWrite,
+	DenyRead,
+	DenyNone
+};
+
+union FileFlagBits
+{
+	uint8_t data = 0;
+	bit_view<0, 3> access_mode;
+	bit_view<4, 3> sharing_mode;
```

Comment:
Well, this still leaves you with bit patterns when you interpret these values 😄 

What I meant:

- Put a link to this page above the union: https://stanislavs.org/helppc/int_21-3d.html
- Add bit-level mappings, naming each bit descriptively, just like I started doing in the VGA header files.

That would make the structure self-describing instead of relying on comments and/or magic bit pattern constants (or hex, which can be even worse).

Reply:
There's not a meaning for a single bit though.  These are both 3 bit patterns, as you can see from that link you just sent.  There's not like a read bit and a write bit.

Reply:
Mea culpa. I did not read it in detail and I'm still half asleep 😅 Ok then, just putting the link there would suffice 😄

Reply:
Added the link